tests/callbacks_test.go

```diff
@@ -601,3 +601,174 @@ func (s *CallbacksSuite) TestNexusResetWorkflowWithCallback() {
 		100*time.Millisecond,
 	)
 }
+
+func blockingWorkflow(ctx workflow.Context) error {
+	return workflow.Await(ctx, func() bool {
+		info := workflow.GetInfo(ctx)
+		return info.OriginalRunID != info.WorkflowExecution.RunID
+	})
+}
+
+func (s *CallbacksSuite) TestNexusResetWorkflowWithCallback_ResetToNotBaseRun() {
+	s.OverrideDynamicConfig(
+		callbacks.AllowedAddresses,
+		[]any{map[string]any{"Pattern": "*", "AllowInsecure": true}},
+	)
+
+	/*
+	 * 1. Start WF w/ no callbacks and immediately terminate
+	 * 2. Start WF second time w/ a callback
+	 * 3. Reset WF back to the first run
+	 * 4. Verify callback is called
+	 */
+
+	tv := testvars.New(s.T())
+	ctx := testcore.NewContext()
+	sdkClient, err := client.Dial(client.Options{
+		HostPort:  s.FrontendGRPCAddress(),
+		Namespace: s.Namespace().String(),
+	})
+	s.NoError(err)
+
+	taskQueue := tv.TaskQueue()
+	workflowID := tv.WorkflowID()
+
+	ch := &completionHandler{
+		requestCh:         make(chan *nexus.CompletionRequest, 2),
+		requestCompleteCh: make(chan error, 2),
+	}
+	callbackAddress := fmt.Sprintf("localhost:%d", freeport.MustGetFreePort())
+	s.runNexusCompletionHTTPServer(s.T(), ch, callbackAddress)
+
+	w := worker.New(sdkClient, taskQueue.GetName(), worker.Options{})
+
+	w.RegisterWorkflow(blockingWorkflow)
+	s.NoError(w.Start())
+	defer w.Stop()
+
+	// 1. Start WF w/ no callbacks and immediately terminate
+	request1 := &workflowservice.StartWorkflowExecutionRequest{
+		RequestId:          uuid.NewString(),
+		Namespace:          s.Namespace().String(),
+		WorkflowId:         workflowID,
+		WorkflowType:       &commonpb.WorkflowType{Name: "blockingWorkflow"},
+		TaskQueue:          taskQueue,
+		Input:              nil,
+		WorkflowRunTimeout: durationpb.New(20 * time.Second),
+		Identity:           s.T().Name(),
+	}
+
+	startResponse1, err := s.FrontendClient().StartWorkflowExecution(ctx, request1)
+	s.NoError(err)
+
+	// Validate the workflow started, then terminate it
+	workflowExecution := &commonpb.WorkflowExecution{
+		WorkflowId: workflowID,
+		RunId:      startResponse1.RunId,
+	}
+	s.WaitForHistoryEvents(`
+			1 WorkflowExecutionStarted
+			2 WorkflowTaskScheduled
+			3 WorkflowTaskStarted
+			4 WorkflowTaskCompleted`,
+		s.GetHistoryFunc(s.Namespace().String(), workflowExecution),
+		5*time.Second,
+		10*time.Millisecond)
+
+	_, err = s.FrontendClient().TerminateWorkflowExecution(ctx, &workflowservice.TerminateWorkflowExecutionRequest{
+		Namespace:         s.Namespace().String(),
+		WorkflowExecution: workflowExecution,
+		Reason:            s.T().Name(),
+		Identity:          tv.WorkerIdentity(),
+	})
+	s.NoError(err)
+
+	s.EqualHistoryEvents(`
```

Comment:
Not required, remove any redundant code from the test to improve readability.
You are not testing `TerminateWorkflowExecution` here, you can assume it works as expected.